google/data_source_google_container_cluster.go

```diff
@@ -0,0 +1,187 @@
+package google
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	container "google.golang.org/api/container/v1"
+)
+
+func dataSourceGoogleContainerCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceContainerClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"additional_zones": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"cluster_ipv4_cidr": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
Except for `name`, `zone`, and `project`, we probably shouldn't allow setting fields since they don't get used to make the request